WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoPressUploadProcessor.swift

```diff
@@ -0,0 +1,174 @@
+        static var playsinline = "playsinline"
+        static var posterUrl = "posterUrl"
+        static var preloadContent = "preloadContent"
+        static var sbc = "sbc"
+        static var sbpc = "sbpc"
+        static var sblc = "sblc"
+        static var useAverageColor = "useAverageColor"
+    }
+
+    init(mediaUploadID: Int32, serverMediaID: Int, videoPressGUID: String) {
+        self.mediaUploadID = mediaUploadID
+        self.serverMediaID = serverMediaID
+        self.videoPressGUID = videoPressGUID
+    }
+
+    lazy var videoPressHtmlProcessor = HTMLProcessor(for: "figure", replacer: { (figure) in
+        var attributes = figure.attributes
+        var html = "<figure "
+        let attributeSerializer = ShortcodeAttributeSerializer()
+        html += attributeSerializer.serialize(figure.attributes)
+        html += "><div class=\"jetpack-videopress-player__wrapper\">"
+        html += "\n\(self.videoPressURL.escapeHtmlNamedEntities())\n"
+        html += "</div></figure>"
+        return html
+    })
+
+    lazy var videoPressBlockProcessor = GutenbergBlockProcessor(for: VideoPressBlockKeys.name, replacer: { videoPressBlock in
+        guard let mediaID = videoPressBlock.attributes[VideoPressBlockKeys.id] as? Int, mediaID == self.mediaUploadID else {
+            return nil
+        }
+        var block = "<!-- \(VideoPressBlockKeys.name) "
+        var attributes = videoPressBlock.attributes
+        attributes[VideoPressBlockKeys.id] = self.serverMediaID
+        attributes[VideoPressBlockKeys.guid] = self.videoPressGUID
+        if let jsonData = try? JSONSerialization.data(withJSONObject: attributes, options: .sortedKeys),
+            let jsonString = String(data: jsonData, encoding: .utf8) {
+            block += jsonString
+        }
+        block += " -->"
+
+        self.videoPressURL = self.getVideoPressURL(attributes)
+        block += self.videoPressHtmlProcessor.process(videoPressBlock.content)
+
+        block += "<!-- /\(VideoPressBlockKeys.name) -->"
+        return block
+    })
+
+    /// The VideoPress URL is built using the same logic we have in Jetpack:
+    /// https://github.com/Automattic/jetpack/blob/b1b826ab38690c5fad18789301ac81297a458878/projects/packages/videopress/src/client/lib/url/index.ts#L19-L67
+    ///
+    /// In order to have a cleaner URL, we only set the options differing from the default VideoPress player settings:
+    /// - Autoplay: Turned OFF by default.
+    /// - Controls: Turned ON by default.
+    /// - Loop: Turned OFF by default.
+    /// - Muted: Turned OFF by default.
+    /// - Plays Inline: Turned OFF by default.
+    /// - Poster: No image by default.
+    /// - Preload: Metadata by default.
+    /// - SeekbarColor: No color by default.
+    /// - SeekbarPlayerColor: No color by default.
+    /// - SeekbarLoadingColor: No color by default.
+    /// - UseAverageColor: Turned ON by default.
+    func getVideoPressURL(_ attributes: [String: Any]) -> String {
```

Comment:
I think it would help to divide this function into smaller pieces. It handles many tasks. We could populate the `options` array in another function and pass it here to populate the URL or something of that sort

Reply:
Good point. It's true that it has to process several options so dividing it up will help readability 👍 .